cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -30,24 +30,36 @@ def __init__(
         bcl_convert_metrics_dir_path: Path,
     ) -> None:
         """Initialize the class."""
-        self.quality_metrics_path: Path = Path(
-            bcl_convert_metrics_dir_path, QUALITY_METRICS_FILE_PATH
-        )
-        self.demux_metrics_path: Path = Path(bcl_convert_metrics_dir_path, DEMUX_METRICS_FILE_PATH)
-        self.adapter_metrics_path: Path = Path(
-            bcl_convert_metrics_dir_path, ADAPTER_METRICS_FILE_PATH
-        )
-        self.sample_sheet_path: Path = Path(bcl_convert_metrics_dir_path, SAMPLE_SHEET_FILE_PATH)
+        self.bcl_convert_demultiplex_dir: Path = bcl_convert_metrics_dir_path
         self.quality_metrics: List[BclConvertQualityMetrics] = self.parse_metrics_file(
-            metrics_file_path=self.quality_metrics_path, metrics_model=BclConvertQualityMetrics
+            metrics_file_path=get_file_in_directory(
+                directory=self.bcl_convert_demultiplex_dir, file_name=QUALITY_METRICS_FILE_NAME
+            ),
+            metrics_model=BclConvertQualityMetrics,
         )
         self.demux_metrics: List[BclConvertDemuxMetrics] = self.parse_metrics_file(
-            metrics_file_path=self.demux_metrics_path, metrics_model=BclConvertDemuxMetrics
+            metrics_file_path=get_file_in_directory(
```

Comment:
I think it is more readable to not nest function calls inside other function calls. So I'd prefer this since it becomes easier to understand each function call and it is not necessary to understand both at once:

```python
metrics_file_path = get_file_in_directory(
    directory=self.bcl_convert_demultiplex_dir, file_name=DEMUX_METRICS_FILE_NAME
)

self.demux_metrics: List[BclConvertDemuxMetrics] = self.parse_metrics_file(
    metrics_file_path=metrics_file_path,
    metrics_model=BclConvertDemuxMetrics,
)
```